marine.c

```diff
@@ -579,6 +579,38 @@ WS_DLL_PUBLIC marine_result *marine_dissect_packet(int filter_id, unsigned char
     return result;
 }
 
+int compile_bpf(char *bpf, struct bpf_program *fcode, char *err_msg) {
+    pcap_t *pc;
+
+    pc = pcap_open_dead(DLT_EN10MB, MIN_PACKET_SIZE);
+    if (pc == NULL) {
+        return -1;
+    }
+    
+    if (pcap_compile(pc, fcode, bpf, 0, 0) == -1) {
```

Comment:
You always call `pcap_close`, so you can simplify everything to:
```c
int compilation_status = pcap_compile(...);
pcap_close(...);
if (compilation_status != 0) {
    // error handling
}
return compilation_status;
```

Reply:
I ended up leaving the code as is because I cannot return `compilation_status` anymore (due to the switch of working with booleans)

Reply:
In my opinion, it *does* make sense to return an error code here.

Reply:
I'm with @yehonatanz on the simplification (however not with him on the error codes - I think we should normalize the result to a boolean since it's not interesting outside of this function).

And then you can do (since you can also remove the error message):
```c
int compilation_status = pcap_compile(...);
pcap_close(...);
return compilation_status == 0;
```